CSC8503/Boss.h

```diff
@@ -101,12 +101,16 @@ namespace NCL
                 }
             }
 
-            BossBullet* releaseBossBullet(Vector3 v, Vector3 s)
+            BossBullet* releaseBossBullet(Vector3 v, Vector3 s, Vector3 p = Vector3{99999,99999,99999})
```

Comment:
I might suggest making use of std::optional here, as it is more clear and robust.

Reply:
Other than that, this seems fine so I'm happy to merge this branch and have this change take place later, if you want to make the change anyway.